src/autofill-utils.js

```diff
@@ -413,6 +413,23 @@ function whenIdle (callback) {
     }
 }
 
+/**
+ * Truncate string from the middle if exceeds the totalLength (default: 30)
+ * @param {string} string
+ * @param {number} totalLength
+ * @returns {string}
+ */
+function truncateFromMiddle (string, totalLength = 30) {
+    if (totalLength < 4) {
+        throw new Error('Do not use with strings shorter than 4')
+    }
```

Comment:
2 or 3 don't make sense, because we add the `…` character when we truncate so there's no practical gain. Returning the string unchanged makes sense, though. I'll do that. 👍

Reply:
Sounds good.
Just to clarify your first point -- the 2 or 3 use case is for the expected truncation length, not the inputted string length, right? Example:
```
> truncateFromMiddle("hello world", 3)
< "he…d"
> truncateFromMiddle("hello world", 2)
< "h…d"
```
Granted, not that I can see much of a use case for very small truncation lengths like this.

Reply:
Sorry, yeah, you're right.